src/Mod/PartDesign/App/FeatureRevolution.cpp

```diff
@@ -138,16 +160,64 @@ App::DocumentObjectExecReturn *Revolution::execute(void)
         for (;xp.More(); xp.Next()) {
             if (checkLineCrossesFace(gp_Lin(pnt, dir), TopoDS::Face(xp.Current())))
                 return new App::DocumentObjectExecReturn("Revolve axis intersects the sketch");
-        }        

```

Comment:
The execute() function is by far too long and should be refactored.

Reply:
Possibly yes. It would also help to make a superclass like pad/pocket. Thing is I still can't wrap my head around the exact algorithm. Should try pen and paper first :grimacing:.

Reply:
@wwmayer I moved some of the logic to separate methods.